### PR TITLE
Fix CDB_DropOverviews and CDB_Overviews problems

### LINKED ISSUE #223: CDB_DropOverviews fails for org users
This bug has been introduced in 0.15.0: https://github.com/CartoDB/cartodb-postgresql/blob/e68d5eca4587d17da4eee9ade260fb4c7eed32c4/scripts-available/CDB_Overviews.sql#L123

Since `row.overview_table` is a regclass it should not be formatted with '%I' and it should not be prependend a schema. The problem will manifest when the overview_table requires the schema 
name to be accessed (i.e. when casting the regclass to text contains the schema part).

The bug is not detected by tests because no tables in schemas that are not in the `search_path` are used in the tests.

### PR DESCRIPTION
**Pull request by jgoizueta**
Fixes #223 and #218

### CODE REVIEW

**Comment by jgoizueta** [comment]
Please @rafatower CR this. We can release this as 0.15.1 to solve all the organization-user-related problems I've detected.

**Comment by jgoizueta** [comment]
cc @ethervoid

File: scripts-available/CDB_Overviews.sql [hunk 1]
```diff
@@ -1,4 +1,30 @@
...
+  FROM pg_class c
+  JOIN pg_namespace n ON n.oid = c.relnamespace
+  WHERE c.relkind = 'r'
+  AND c.relname NOT IN ('cdb_tablemetadata', 'spatial_ref_sys')
+  AND CASE WHEN schema_name IS NULL
+             THEN n.nspname NOT IN ('pg_catalog', 'information_schema', 'topology', 'cartodb')
+           ELSE n.nspname = schema_name END
+  AND CASE WHEN perm = 'public' THEN has_table_privilege('publicuser', c.oid, 'SELECT')
+           WHEN perm = 'private' THEN has_table_privilege(current_user, c.oid, 'SELECT') AND NOT has_table_privilege('publicuser', c.oid, 'SELECT')
+           WHEN perm = 'all' THEN has_table_privilege(current_user, c.oid, 'SELECT') OR has_table_privilege('publicuser', c.oid, 'SELECT')
```
Comment (rafatower) [inline scripts-available/CDB_Overviews.sql]: Minor comment: 

I understand that this is basiclally copied from CDB_UserTables with some tweaks. But I also understand that if the user has perms on the base table then they should be able to perform any operation on them regardless of the perms in overviews tables.

Therefore the code to check privileges could be removed? the less code the less maintenance.

But as it is tested, ok, go ahead with it :)

File: scripts-available/CDB_Overviews.sql [hunk 2]
```diff
@@ -140,16 +166,15 @@ RETURNS TABLE(base_table REGCLASS, z integer, overview_table REGCLASS)
...
-    -- TODO: review implementation of CDB_UserTables an suitability for this
-    SELECT * FROM _cdb_split_table_name(reloid) INTO schema_name, table_name;
+    SELECT * FROM _cdb_split_table_name(reloid) INTO schema_name, _table_name;
     RETURN QUERY SELECT
       reloid AS base_table,
-      _CDB_OverviewTableZ(cdb_usertables) AS z,
-      ('"' || schema_name|| '"."' ||cdb_usertables || '"')::regclass AS overview_table
-      FROM CDB_UserTables()
-      WHERE _CDB_IsOverviewTableOf((SELECT relname FROM pg_class WHERE oid=reloid), cdb_usertables)
+      _CDB_OverviewTableZ(table_name) AS z,
```
Comment (rafatower) [inline scripts-available/CDB_Overviews.sql]: there's a `table_name` and a `_table_name` (prefixed with underscore). Don't know when the one w/o underscore is declared.

if it is declared, they have bad naming making the code harder to understand

File: scripts-available/CDB_Overviews.sql [hunk 3]
```diff
@@ -204,7 +231,7 @@ AS $$
         -- This is the typical ERROR: stats for "mytable" do not exist
         WHEN internal_error THEN
           -- Get stats and execute again
-          EXECUTE format('ANALYZE %1$I', reloid);
+          EXECUTE format('ANALYZE %1$s', reloid);
```
Comment (rafatower) [inline scripts-available/CDB_Overviews.sql]: what's this ANALYZE for?

File: scripts-available/CDB_Overviews.sql (hunk 2 above)
Comment (jgoizueta) [inline scripts-available/CDB_Overviews.sql]: Not a bug but confusing naming: the one with underscore is variable used to get the second result of `_cdb_split_table_name` which isn't actually used; the other identifer is a column of `_CDB_UserTablesInSchema`. Should change the name of the first to make more evident it not being used.

File: scripts-available/CDB_Overviews.sql (hunk 1 above)
Comment (jgoizueta) [inline scripts-available/CDB_Overviews.sql]: I agree to remove it

File: scripts-available/CDB_Overviews.sql (hunk 3 above)
Comment (jgoizueta) [inline scripts-available/CDB_Overviews.sql]: I hoped you knew! (you placed that code there). Is it supposed to generate the stats, the lack of which could have provoked an error?

Comment (rafatower) [inline scripts-available/CDB_Overviews.sql]: :dancer: now I remember: when using the quick extend function, there's little chance of error if it doesn't have the stats. unlikely to happen.

**Comment by iriberri** [comment]
@jgoizueta I just pushed the changes for the bugs we found while testing. I a brief description of them in the commit description.

**Comment by jgoizueta** [comment]
:+1:

**Pull request merged**